### DISCUSSION TITLE
Checking Integrity and Consistency

### DISCUSSION THREAD

**Discussion by MichaelEPope** [cupcakearmy/autorestic#239] [General]
Hey there,

This is a really awesome tool, so thank you for that :).

Is there an easy way to do checks for integrity and consistency, as described here:  https://restic.readthedocs.io/en/stable/045_working_with_repos.html?highlight=check#checking-integrity-and-consistency

In particular, I'm interested in --read-data-subset=2%, if anybody has seen that work.  If there isn't a way, no worries, I can use a regular restic command, but I thought I'd ask :).

I'm guessing I need to use exec though?

**Comment by cupcakearmy** [comment]
`autorestic check` already shoudl do that. 
However if you want to it manually you can use `autorestic exec -b my-backend -v -- check --read-data-substet=69%`
